zone/spells.cpp

```diff
@@ -4548,39 +4548,62 @@ bool Mob::SpellOnTarget(
 	return true;
 }
 
-void Corpse::CastRezz(uint16 spellid, Mob* Caster)
+void Corpse::CastRezz(uint16 spell_id, Mob* Caster)
 {
-	LogSpells("Corpse::CastRezz spellid [{}], Rezzed() is [{}], rezzexp is [{}]", spellid,IsRezzed(),rez_experience);
-
-	if(IsRezzed()){
-		if(Caster && Caster->IsClient())
-			Caster->Message(Chat::Red,"This character has already been resurrected.");
-
+	LogSpells("Corpse::CastRezz spell_id [{}], Rezzed() is [{}], rezzexp is [{}], rez_timer enabled:: [{}]", spell_id,IsRezzed(),rez_experience, corpse_rez_timer.Enabled());
+
+	// refresh rezzed state from database
+	uint32 db_exp, db_gm_exp;
+	bool db_rezzable, db_is_rezzed;
+	if (!database.LoadCharacterCorpseRezData(corpse_db_id, &db_exp, &db_gm_exp, &db_rezzable, &db_is_rezzed)) {
+		// db error, corpse disappeared?
+		Caster->MessageString(Chat::White, REZZ_ALREADY_PENDING);
 		return;
 	}
-	/*
-	if(!can_rez) {
-		if(Caster && Caster->IsClient())
+
+	rezzable = db_rezzable;
+	IsRezzed(db_is_rezzed);
+	rez_experience = db_exp;
+	gm_rez_experience = db_gm_exp;
+
+	// Rez timer has expired, only GMs can rez at this point. (uses rezzable)
+	if (!IsRezzable()) {
+		if (Caster && Caster->IsClient() && !Caster->CastToClient()->GetGM()) {
+			Caster->MessageString(Chat::White, REZZ_ALREADY_PENDING);
 			Caster->MessageString(Chat::White, CORPSE_TOO_OLD);
```

Comment:
Looks like you're sending both messages here?

Reply:
Correct, both messages get sent in this case. (This code is based on TAKPs Implementation and messages being sent).